workspace_host/interface.js

```diff
@@ -106,64 +108,104 @@ app.post('/', function(req, res) {
         resetSequence(workspace_id, res);
     } else if (action == 'destroy') {
         destroySequence(workspace_id, res);
+    } else if (action == 'grade') {
+        gradeSequence(workspace_id, res);
     } else if (action == 'status') {
         res.status(200).send('Running');
     } else {
         res.status(500).send(`Action '${action}' undefined`);
     }
 });
 
+
+
 app.listen(config.workspaceDevHostPort, () => console.log(`Listening on port ${config.workspaceDevHostPort}`));
 
 // For detecting file changes
-var update_queue = {};  // key: path of file on local, value: action ('update' or 'remove').
+class jobQueue {
```

Comment:
It looks like this PR includes the changes from #2834. I don't want to merge #2834 yet, so can you remove these changes from this PR so that this one only includes the `grade/` endpoint addition?

Reply:
Removed.